src/figuresClasses.ts

```diff
@@ -1,11 +1,93 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string;
+
+  color: string;
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    if (Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
```

Comment:
The condition for checking if the sides can form a triangle is incorrect. The correct condition should be: `if (a + b <= c || a + c <= b || b + c <= a)`. This ensures that the sum of any two sides is greater than the third side, which is the triangle inequality theorem.